src/pymor/operators/numpy.py

```diff
@@ -418,59 +420,131 @@ def _format_repr(self, max_width, verbosity):
         return super()._format_repr(max_width, verbosity, override={'matrix': matrix_repr})
 
 
-class NumpyHankelOperator(NumpyGenericOperator):
-    r"""Implicit representation of a Hankel operator by a |NumPy Array|.
+class NumpyCirculantOperator(Operator, CacheableObject):
+    r"""Matrix-free representation of a circulant matrix by a |NumPy Array|.
 
-    Let
+    A circulant matrix is a special kind of Toeplitz matrix which is square and completely
+    determined by its first column via
 
     .. math::
-        h =
-        \begin{pmatrix}
-            h_1 & h_2 & \dots & h_n
-        \end{pmatrix},\quad h_i\in\mathbb{C}^{p\times m},\,i=1,\,\dots,\,n,\quad n,m,p\in\mathbb{N}
+        C =
+            \begin{bmatrix}
+                c_1    & c_n    & c_{n-1} & \cdots & \cdots & c_2 \\
+                c_2    & c_1    & c_n     &        &        & \vdots \\
+                c_3    & c_2    & \ddots  &        &        & \vdots \\
+                \vdots &        &         & \ddots & c_n    & c_{n-1} \\
+                \vdots &        &         & c_2    & c_1    & c_n \\
+                c_n    & \cdots & \cdots  & c_3    & c_2    & c_1
+            \end{bmatrix} \in \mathbb{C}^{n \times n},
+
+    where the so-called circulant vector :math:`c \in \mathbb{C}^n` denotes the first column of the
+    matrix. The matrix :math:`C` as seen above is not explicitly constructed, only `c` is stored.
+    Efficient matrix-vector multiplications are realized with DFT in the class' `apply` method.
+    See :cite:`GVL13` Chapter 4.8.2. for details.
 
-    be a finite (matrix-valued) sequence. If :math:`h` has an odd number :math:`n=2s-1`
-    of elements, the corresponding Hankel operator can be represented by the matrix
+    Parameters
+    ----------
+    c
+        The |NumPy array| that defines the circulant vector.
```

Comment:
Can this class handle `c` as a 3D array?

Reply:
yes!